common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -212,23 +233,7 @@ public MicrosoftStsAccount createAccount(
 
         MicrosoftStsAccount account = new MicrosoftStsAccount(idToken, clientInfo);
 
-        // The Account created by the strategy sets the environment to get the 'iss' from the IdToken
-        // For caching purposes, this may not be the correct value due to the preferred cache identifier
-        // in the InstanceDiscoveryMetadata
-        URL authority = null;
-
-        try {
-            authority = new URL(mTokenEndpoint);
-        } catch (MalformedURLException e) {
-            Logger.verbose(
-                    TAG + methodName,
-                    "Creating account from TokenResponse failed due to malformed URL (mTokenEndpoint)..."
-            );
-        }
-
-        if (authority != null) {
-            account.setEnvironment(getIssuerCacheIdentifierFromAuthority(authority));
-        }
+        account.setEnvironment(getIssuerCacheIdentifierFromTokenEndpoint());
```

Comment:
should we use this as a single source of truth?

Reply:
I've fixed this in MSAL, we will now get cache identifier from token endpoint in MSALOAuth2TokenCache. However, I am not going to remove the `getIssuerCacheIdentifier(AuthorizationRequest)` method as that is an abstract method and is being overridden by a few different classes.